### PR TITLE
Windows command line parsing fix

### PR DESCRIPTION
**Pull request by beardypig**
This PR does a couple of things. 

Firstly, it fixes the issue reported by @bastimeyer in #293, with some tests (see below). `subprocess.Popen` can be called with a string for the command line under Windows (same as `shell=True` under POSIX). Calling `subprocess.Popen` with the string command line means that it doesn't have to be parsed using `shlex` or quoted, which avoids the parsing issue. Parsing is still required on POSIX as `shell=True` is not advised. 

I tested in `cmd` and `powershell` on Vista and Windows 10:
```powershell
streamlink -a "--input-title-format \"Poker \\\"Stars\\\"\" {filename}" twitch.tv/pokerstars worst
streamlink -p "C:\Program Files\VideoLAN\VLC\vlc.exe --input-title-format \"Poker \\\"Stars\\\"\"" twitch.tv/pokerstars worst
streamlink twitch.tv/pokerstars worst
```
Also, tested in `streamlink-twitch-gui`, I disabled the custom window title and then added `--input-title-format "\"{name}\" {game}"` to the custom parameters. VLC displayed `"Stream Name" Game` in the title :)

Secondly, it adds some automated testing on Windows via AppVeyor, the test script only installs and runs the tests. I moved the dev/test dependencies to `dev-requirements.txt` so they could be shared between Travis and AppVeyor easily. It might need some extra tweaking after the merge though :)

### CODE REVIEW

**Comment by gravyboat** [comment]
Nice @beardypig! This looks good to me. I'm going to leave it open in case anyone else has feedback (especially @bastimeyer).

File: tests/test_cmdline.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        if not passthrough:
+            mock_popen.assert_called_with(commandline, stderr=ANY, stdout=ANY, bufsize = ANY, stdin = ANY)
+        else:
+            mock_popen.assert_called_with(commandline, stderr=ANY, stdout=ANY)
+
+    #
+    # POSIX tests
+    #
+
+    @unittest.skipIf(is_win32, "test only applicable in a POISX OS")
```
Comment (bastimeyer) [inline tests/test_cmdline.py]: Found a typo 😅 
Also the copied lines below...

Comment (beardypig) [inline tests/test_cmdline.py]: It actually refers to a French OS PoisX ...

**Comment by gravyboat** [comment]
Good catch I totally missed that, merging! Thanks for reviewing @bastimeyer!

**Pull request merged**